std/algorithm/comparison.d

```diff
@@ -1298,6 +1298,62 @@ MaxType!T max(T...)(T args)
     return cast(typeof(return)) (chooseB ? b : a);
 }
 
+/**
+Iterates the passed range and selects the lowest value matching $(D pred).
+Can use a custom key getter, as defined by $(D keyGetter)
+(which is the identity by default).
+
+Params:
+    r = Range from which the minimum will be selected
+
+Returns: The lowest value according to $(D pred) of the passed-in values.
+*/
+auto selectOne(alias pred, alias keyGetter = (a) => a, Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias predFun = binaryFun!pred;
+    alias keyGetterFun = unaryFun!keyGetter;
+    auto bestValue = keyGetterFun(r.front);
+    ElementType!Range bestEl = r.front;
+    r.popFront();
+    foreach (el; r)
```

Comment:
Code now uses `while(r.empty)` for forward ranges and an index based loop for arrays - is there even a better way?